### PR TITLE
Added triage observable analyzer

### LINKED ISSUE #142: Triage analyzer
We could create a file analyzers that leverages Triage Sandbox. https://tria.ge
Recently, they added the chance to request a free "researcher" account that can leverage their API to programmatically start sandbox analysis and retrieve results. https://tria.ge/docs/cloud-api/

### PR DESCRIPTION
**Pull request by m0mosenpai**
# Description

Added triage analyzer for scanning observables - searching through existing reports via URL or Hash query; or submitting a sample for analysis via a URL. Also added documentation for both triage observable and file analyzers.

## Fixes
closes #142 

## Type of change
- [x] New feature (non-breaking change which adds functionality).

# Checklist

- [x] The pull request is for the branch develop
- [x] I changed the file [Usage](https://github.com/intelowlproject/IntelOwl/blob/master/docs/source/Usage.md) and [ReadMe](https://github.com/intelowlproject/IntelOwl/blob/master/README.md) if I added a new analyzer. 
- [x] I have added tests in the [Tests](https://github.com/intelowlproject/IntelOwl/blob/master/tests) folder. 
- [x] The tests gave 0 errors.
- [x] `Black` gave 0 errors.
- [x] `Flake` gave 0 errors.
- [x] I squashed the commits into a single one.

# Real World Example
1. `submit` config paramenter - submitting file via url (similar report generated as triage file analyzer)
image

2. `search` config parameter - searching for reports via hash.
image
50 results (default) (max 200) in same format as above followed by a `next` key. I haven't implemented pagination using `next` as of now. A bit confused on how to go about it since next contains a date-time stamp of sorts. 
image

3. `search` config parameter - searching for reports via URL.
I couldn't find a good example of this (a malicious url) but report should be similar to the hash case.

### CODE REVIEW

File: api_app/script_analyzers/observable_analyzers/triage_search.py [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+        self.api_key_name = additional_config_params.get("api_key_name", "TRIAGE_KEY")
+        self.__api_key = secrets.get_secret(self.api_key_name)
+        self.analysis_type = additional_config_params.get("analysis_type", "search")
+        self.report_type = additional_config_params.get("report_type", "overview")
+        self.max_tries = additional_config_params.get("max_tries", 1)
+        self.poll_distance = 3
+
+    def run(self):
+        if not self.__api_key:
+            raise AnalyzerRunException(
```
Comment (Eshaan7) [inline api_app/script_analyzers/observable_analyzers/triage_search.py:32]: should be [`AnalyzerConfigurationException`](https://github.com/intelowlproject/IntelOwl/blob/fa472db8c9b3d4c7ec5201ec552f6f928b076814/api_app/exceptions.py#L5)

File: api_app/script_analyzers/observable_analyzers/triage_search.py [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+
+        self.headers = {"Authorization": f"Bearer {self.__api_key}"}
+
+        response = None
+        if self.analysis_type == "search":
+            response = self.__triage_search()
+        elif self.analysis_type == "submit":
+            response = self.__triage_submit()
+        else:
+            raise AnalyzerRunException("unknown analysis type")
```
Comment (Eshaan7) [inline api_app/script_analyzers/observable_analyzers/triage_search.py:44]: should be [`AnalyzerConfigurationException`](https://github.com/intelowlproject/IntelOwl/blob/fa472db8c9b3d4c7ec5201ec552f6f928b076814/api_app/exceptions.py#L5).

The error message should be something similar to [this](https://github.com/intelowlproject/IntelOwl/blob/1d1d61fd318c686776848d58733e6877d39b5ee1/api_app/script_analyzers/observable_analyzers/shodan.py#L32-L33).

File: configuration/analyzer_config.json [hunk 3]
```diff
@@ -955,6 +955,23 @@
...
+    ],
+    "description": "analyze a url sample using triage sandbox environment",
+    "python_module": "triage_search.TriageSearch",
+    "requires_configuration": true,
+    "additional_config_params": {
+      "api_key_name": "TRIAGE_KEY",
+      "endpoint": "public",
+      "analysis_type": "search",
+      "report_type": "overview",
+      "soft_time_limit": 300
```
Comment (Eshaan7) [inline configuration/analyzer_config.json:972]: `soft_time_limit` should be outside of of `additional_config_param`

File: tests/test_observables.py [hunk 4]
```diff
@@ -543,3 +564,15 @@ def test_cymru_get(self, mock_get=None, mock_post=None):
             {},
         ).start()
         self.assertEqual(report.get("success", False), True)
+
+    @mock_connections(patch("requests.get", side_effect=mocked_triage_get))
+    @mock_connections(patch("requests.post", side_effect=mocked_triage_post))
+    def test_triage_search(self, *args):
```
Comment (Eshaan7) [inline tests/test_observables.py:570]: should duplicate this test for the case `test_triage_submit` passing `"{ analysis_type": "submit" }` as `additional_config_param`

File: tests/test_observables.py [hunk 5]
```diff
@@ -505,6 +514,18 @@ def test_robtex_fdns(self, mock_get=None, mock_post=None):
         ).start()
         self.assertEqual(report.get("success", False), True)
 
+    @mock_connections(patch("requests.get", side_effect=mocked_triage_get))
+    @mock_connections(patch("requests.post", side_effect=mocked_triage_post))
+    def test_triage_search(self, *args):
```
Comment (Eshaan7) [inline tests/test_observables.py:519]: should duplicate this test for the case `test_triage_submit` passing `"{ analysis_type": "submit" }`

**Comment by mlodic** [comment]
Please `git rebase develop` to get recent changes on Continuous Integration management (Travis dropped support for OS projects so we had to move on). In this way, checks will trigger again with the new configuration. Thank you

**Review by Eshaan7 — CHANGES_REQUESTED** [review]

File: configuration/analyzer_config.json (hunk 3 above)
Comment (Eshaan7) [reply configuration/analyzer_config.json:972]: please make same change on line number 985

**Comment by m0mosenpai** [comment]
Please let me know if there's anything else to change else I'll squash the commits.

File: configuration/analyzer_config.json [hunk 6]
```diff
@@ -994,6 +994,22 @@
...
+      "url",
+      "hash"
+    ],
+    "description": "analyze a url sample using triage sandbox environment",
+    "python_module": "triage_search.TriageSearch",
+    "requires_configuration": true,
+    "additional_config_params": {
+      "api_key_name": "TRIAGE_KEY",
+      "endpoint": "public",
+      "report_type": "overview"
```
Comment (Eshaan7) [inline configuration/analyzer_config.json:1009]: should add `analysis_type: "submit"` parameter. Explicit is better than implicit.

File: configuration/analyzer_config.json [hunk 7]
```diff
@@ -1003,9 +1019,9 @@
     "additional_config_params": {
       "api_key_name": "TRIAGE_KEY",
       "endpoint": "public",
```
Comment (Eshaan7) [inline configuration/analyzer_config.json:1021]: should add `analysis_type: "search"` parameter. Explicit is better than implicit.

File: configuration/analyzer_config.json [hunk 8]
```diff
@@ -994,6 +994,22 @@
     "python_module": "tor.Tor",
     "soft_time_limit": 30
   },
+  "Triage_Search": {
+    "type": "observable",
+    "observable_supported": [
+      "url",
+      "hash"
+    ],
+    "description": "analyze a url sample using triage sandbox environment",
```
Comment (Eshaan7) [inline configuration/analyzer_config.json:1003]: add another key-value `"external_service": true,`

File: configuration/analyzer_config.json [hunk 9]
```diff
@@ -994,6 +994,22 @@
...
+    "requires_configuration": true,
+    "additional_config_params": {
+      "api_key_name": "TRIAGE_KEY",
+      "endpoint": "public",
+      "report_type": "overview"
+    },
+    "soft_time_limit": 300
+  },
   "Triage_Scan": {
     "type": "file",
```
Comment (Eshaan7) [inline configuration/analyzer_config.json:1014]: add another key-value `"external_service": true,`

**Pull request merged**